pkg/trace/test/testsuite/traces_test.go

```diff
@@ -89,26 +90,46 @@ func TestTraces(t *testing.T) {
 	})
 
 	t.Run("filter_tags", func(t *testing.T) {
-		if err := r.RunAgent([]byte("apm_config:\r\n  filter_tags.require: [env:prod, db:mysql]\r\n  filter_tags.reject: [outcome:success]")); err != nil {
+		if err := r.RunAgent([]byte(`apm_config:
+  filter_tags
+    require:["env:prod", "db:mysql"]
+    reject:["outcome:success"]`)); err != nil {
 			t.Fatal(err)
 		}
 		defer r.KillAgent()
 
-		p := testutil.GeneratePayload(5, &testutil.TraceConfig{
-			MinSpans: 5,
+		pGood := testutil.GeneratePayload(4, &testutil.TraceConfig{
+			MinSpans: 4,
 			Keep:     true,
 		}, nil)
-		for _, span := range p[2] {
+		for _, span := range pGood[2] {
 			span.Meta = map[string]string{
-				"env": "prod",
-				"db":  "mysql",
+				"outcome": "success",
+				"db":      "mysql",
 			}
 		}
-		if err := r.Post(p); err != nil {
+
+		pBad := testutil.GeneratePayload(4, &testutil.TraceConfig{
+			MinSpans: 4,
+			Keep:     true,
+		}, nil)
+		for _, span := range pBad[2] {
+			span.Meta = map[string]string{
+				"outcome": "success",
+			}
+		}
+
+		if err := r.Post(pGood); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := r.Post(pBad); err != nil {
 			t.Fatal(err)
 		}
+
 		waitForTrace(t, &r, func(v pb.TracePayload) {
-			payloadsEqual(t, append(p[:2], p[3:]...), v)
+			assert.Equal(t, 4, len(pGood))
+			assert.Equal(t, 4, len(pBad))
```

Comment:
Here also, we are verifying the length of the input, not the output. The output is `v`. We have to check that inside `v` we have all of the traces except the filtered ones.